modules/chat/src/routes/InvitationRoutes.ts

```diff
@@ -0,0 +1,243 @@
+
+  declareRoutes() {
+    this.routingManager.route(
+      {
+        path: '/hook/:answer/:invitationToken',
+        action: ConduitRouteActions.GET,
+        urlParams: {
+          answer: ConduitString.Required,
+          invitationToken: ConduitString.Required,
+        },
+      },
+      new ConduitRouteReturnDefinition('InvitationResponse', 'String'),
+      this.answerInvitationFromHook.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/invitation/:answer/:id',
+        action: ConduitRouteActions.GET,
+        urlParams: {
+          id: ConduitString.Required,
+          answer: ConduitString.Required,
+        },
+        middlewares: ['authMiddleware'],
+      },
+      new ConduitRouteReturnDefinition('InvitationResponse', 'String'),
+      this.answerInvitation.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/invitations/received',
+        action: ConduitRouteActions.GET,
+        queryParams: {
+          skip: ConduitNumber.Optional,
+          limit: ConduitNumber.Optional,
+        },
+        middlewares: ['authMiddleware'],
+      },
+      new ConduitRouteReturnDefinition('GetInvitationsResponse', {
+        invitations: [InvitationToken.getInstance().fields],
+        count: ConduitNumber.Required,
+      }),
+      this.getInvitations.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/invitations/sent',
+        queryParams: {
+          skip: ConduitNumber.Optional,
+          limit: ConduitNumber.Optional,
+        },
+        action: ConduitRouteActions.GET,
+        middlewares: ['authMiddleware'],
+      },
+      new ConduitRouteReturnDefinition('SentInvitationsResponse', {
+        invitations: [InvitationToken.getInstance().fields],
+        count: ConduitNumber.Required,
+
+      }),
+      this.sentInvitations.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/invitation/cancel/:id',
+        action: ConduitRouteActions.DELETE,
+        urlParams: {
+          id: ConduitString.Required,
+        },
+        middlewares: ['authMiddleware'],
+      },
+      new ConduitRouteReturnDefinition('InvitationCancelResponse', 'String'),
+      this.cancelInvitation.bind(this),
+    );
+
+  }
+
+  async answerInvitation(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { id, answer } = call.request.params;
+    const { user } = call.request.context;
+    const invitationTokenDoc: InvitationToken | null = await InvitationToken.getInstance().findOne({
+      _id: id,
+      receiver: user._id,
+    });
+    if (isNil(invitationTokenDoc)) {
+      throw new GrpcError(status.NOT_FOUND, 'Invitation not valid');
+    }
+    let message;
+    let receiver = user._id;
+    const accepted = (answer === 'accept');
+    const chatRoom = await ChatRoom.getInstance().findOne({ _id: invitationTokenDoc.room });
+    if (isNil(chatRoom)) {
+      throw new GrpcError(status.NOT_FOUND, 'Room not exist');
```

Comment:
Still here. Error message should be `Room does not exist`.
Ideally keep this consistent across all routes (answerInvitationFromHook responds with `Chat room does not exist`)